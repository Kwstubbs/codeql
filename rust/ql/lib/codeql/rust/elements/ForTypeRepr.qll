// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
