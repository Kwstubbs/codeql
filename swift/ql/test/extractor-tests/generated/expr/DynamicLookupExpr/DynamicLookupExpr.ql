// generated by codegen/codegen.py, do not edit
import codeql.swift.elements
import TestUtils

query predicate instances(
  DynamicLookupExpr x, string primaryQlClasses, string hasType__label, string hasType,
  string getBase__label, Expr getBase, string hasMember__label, string hasMember
) {
  toBeTested(x) and
  not x.isUnknown() and
  primaryQlClasses = x.getPrimaryQlClasses() and
  hasType__label = "hasType:" and
  (if x.hasType() then hasType = "yes" else hasType = "no") and
  getBase__label = "getBase:" and
  getBase = x.getBase() and
  hasMember__label = "hasMember:" and
  if x.hasMember() then hasMember = "yes" else hasMember = "no"
}

query predicate getType(DynamicLookupExpr x, Type getType) {
  toBeTested(x) and not x.isUnknown() and getType = x.getType()
}

query predicate getMember(DynamicLookupExpr x, Decl getMember) {
  toBeTested(x) and not x.isUnknown() and getMember = x.getMember()
}
