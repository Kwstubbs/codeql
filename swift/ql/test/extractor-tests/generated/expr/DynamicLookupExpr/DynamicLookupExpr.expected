instances
| dynamic_lookup.swift:15:1:15:3 | .foo(_:) | DynamicMemberRefExpr | hasType: | yes | getBase: | dynamic_lookup.swift:15:1:15:1 | OpaqueValueExpr | hasMember: | yes |
| dynamic_lookup.swift:16:5:16:9 | subscript ...[...] | DynamicSubscriptExpr | hasType: | yes | getBase: | dynamic_lookup.swift:16:5:16:5 | OpaqueValueExpr | hasMember: | yes |
getType
| dynamic_lookup.swift:15:1:15:3 | .foo(_:) | ((Int) -> ())? |
| dynamic_lookup.swift:16:5:16:9 | subscript ...[...] | Int? |
getMember
| dynamic_lookup.swift:15:1:15:3 | .foo(_:) | dynamic_lookup.swift:6:9:6:28 | foo(_:) |
| dynamic_lookup.swift:16:5:16:9 | subscript ...[...] | dynamic_lookup.swift:7:9:9:3 | subscript ... |
